File: 1.Two_Sum/1.Two_Sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hash = HashMap<Int, Int>()
        for (i in 0 until nums.size) {
            if (hash.containsKey(target - nums[i]))
                return intArrayOf(hash.get(target - nums[i]) as Int, i)
            else
                hash.put(nums[i], i)
        }
        return intArrayOf()
    }
}

File: 1.Two_Sum/1.Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hash = [Int : Int]()
        for i in 0...nums.count {
            if (hash[target - nums[i]] != nil) {
                return [hash[target - nums[i]]!, i]
            }
            else {
                hash[nums[i]] = i
            }
        }
        return []
    }
}
